app/src/main/java/nerd/tuxmobil/fahrplan/congress/autoupdate/UpdateService.kt

```diff
@@ -0,0 +1,149 @@
+package nerd.tuxmobil.fahrplan.congress.autoupdate
+
+import android.app.PendingIntent
+import android.app.PendingIntent.FLAG_IMMUTABLE
+import android.app.PendingIntent.FLAG_ONE_SHOT
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.text.TextUtils
+
+import androidx.core.app.NotificationCompat
+import androidx.core.app.SafeJobIntentService
+
+import java.util.concurrent.CountDownLatch
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+import nerd.tuxmobil.fahrplan.congress.MyApp
+import nerd.tuxmobil.fahrplan.congress.R
+import nerd.tuxmobil.fahrplan.congress.models.Session
+import nerd.tuxmobil.fahrplan.congress.net.ConnectivityObserver
+import nerd.tuxmobil.fahrplan.congress.net.FetchScheduleResult
+import nerd.tuxmobil.fahrplan.congress.net.HttpStatus
+import nerd.tuxmobil.fahrplan.congress.net.LoadShiftsResult
+import nerd.tuxmobil.fahrplan.congress.net.ParseResult
+import nerd.tuxmobil.fahrplan.congress.net.ParseScheduleResult
+import nerd.tuxmobil.fahrplan.congress.net.ParseShiftsResult
+import nerd.tuxmobil.fahrplan.congress.notifications.NotificationHelper
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
+import nerd.tuxmobil.fahrplan.congress.schedule.MainActivity
+import nerd.tuxmobil.fahrplan.congress.utils.FahrplanMisc
+
+class UpdateService : SafeJobIntentService() {
+
+    companion object {
+        private const val JOB_ID = 2119
+        private const val LOG_TAG = "UpdateService"
+        @JvmStatic
+        fun start(context: Context) {
+            enqueueWork(context, UpdateService::class.java, JOB_ID, Intent())
+        }
+    }
+
+    private val appRepository: AppRepository = AppRepository
```

Comment:
The type can be removed.

Reply:
![image](https://user-images.githubusercontent.com/144518/186632481-9abeb589-b2c1-46c4-bc1d-c344450f53b1.png)